seatunnel-engine/seatunnel-engine-core/src/main/java/org/apache/seatunnel/engine/core/dag/actions/ShuffleConfig.java

```diff
@@ -0,0 +1,205 @@
+    public static final int DEFAULT_QUEUE_BACKUP_COUNT = 0;
+    public static final int DEFAULT_QUEUE_ASYNC_BACKUP_COUNT = 0;
+    public static final int DEFAULT_BATCH_SIZE = 1024;
+    public static final long DEFAULT_BATCH_FLUSH_INTERVAL = TimeUnit.SECONDS.toMillis(3);
+
+    private int batchSize;
+    private long batchFlushInterval;
+    private ShuffleStrategy shuffleStrategy;
+
+    @Getter
+    @Setter
+    @AllArgsConstructor
+    @NoArgsConstructor
+    @ToString
+    public abstract static class ShuffleStrategy implements Serializable {
+        private long jobId;
+        private int inputPartitions;
+        private int targetPartitions;
+        private QueueConfig queueConfig;
+
+        public abstract Map<String, IQueue<Record<?>>> createShuffles(HazelcastInstance hazelcast, int inputIndex);
+
+        public abstract String createShuffleKey(Record<?> record, int inputIndex);
+
+        public abstract IQueue<Record<?>>[] getShuffles(HazelcastInstance hazelcast, int targetIndex);
+
+        protected IQueue<Record<?>> getIQueue(HazelcastInstance hazelcast, String queueName) {
+            QueueConfig targetQueueConfig = hazelcast.getConfig().getQueueConfig(queueName);
+            if (queueConfig != null) {
+                targetQueueConfig.setMaxSize(queueConfig.getMaxSize())
+                    .setBackupCount(queueConfig.getBackupCount())
+                    .setAsyncBackupCount(queueConfig.getAsyncBackupCount())
+                    .setEmptyQueueTtl(queueConfig.getEmptyQueueTtl());
+            }
+            return hazelcast.getQueue(queueName);
+        }
+    }
+
+    @NoArgsConstructor
+    @ToString
+    public static class PartitionShuffleStrategy extends ShuffleStrategy {
+        private final Map<Integer, String[]> inputQueueMapping = new HashMap<>();
+
+        public PartitionShuffleStrategy(long jobId,
+                                        int inputPartitions,
+                                        int targetPartitions) {
+            this(jobId, inputPartitions, targetPartitions, null);
+        }
+
+        public PartitionShuffleStrategy(long jobId,
+                                        int inputPartitions,
+                                        int targetPartitions,
+                                        QueueConfig queueConfig) {
+            super(jobId, inputPartitions, targetPartitions, queueConfig);
+        }
+
+        @Override
+        public Map<String, IQueue<Record<?>>> createShuffles(HazelcastInstance hazelcast, int inputIndex) {
+            checkArgument(inputIndex >= 0 && inputIndex < getInputPartitions());
+            Map<String, IQueue<Record<?>>> shuffleMap = new HashMap<>();
+            for (int targetIndex = 0; targetIndex < getTargetPartitions(); targetIndex++) {
+                String queueName = generateQueueName(inputIndex, targetIndex);
+                IQueue<Record<?>> queue = getIQueue(hazelcast, queueName);
+                shuffleMap.put(queueName, queue);
```

Comment:
Maybe should call `clear` before put it into map

Reply:
fixed